src/main/java/com/puppycrawl/tools/checkstyle/checks/header/AbstractHeaderCheck.java

```diff
@@ -113,7 +113,7 @@ public void setHeaderFile(URI uri) throws CheckstyleException {
     private void loadHeaderFile() throws CheckstyleException {
         checkHeaderNotInitialized();
         try (Reader headerReader = new InputStreamReader(new BufferedInputStream(
-                    headerFile.toURL().openStream()), charset)) {
+                    headerFile.toURL().openStream()), Charset.forName(charset))) {
```

Comment:
The setter calls `Charset.isSupported`. Does it make more sense to change the property type to `Charset` and to the conversion in the setter?

Reply:
I thought about it, but decided to keep it simple.
I changed it now